### PR TITLE
Roll Dart to 93d8c9fe2a2c22dc95ec85866af108cfab71ad06.

### PR DESCRIPTION
**Pull request by sivachandra**

### CODE REVIEW

**Comment by sivachandra** [comment]
This PR includes @mraleph 's PR he has shared with me: https://github.com/flutter/engine/pull/4541

**Comment by aam** [comment]
#4541 changes LGTM

File: runtime/dart_init.cc [hunk 1]
```diff
@@ -246,14 +247,15 @@ Dart_Isolate ServiceIsolateCreateCallback(const char* script_uri,
 
   bool is_running_from_kernel = GetKernelPlatformBinary() != nullptr;
 
+  flags->load_vmservice_library = true;
   Dart_Isolate isolate =
       is_running_from_kernel
           ? Dart_CreateIsolateFromKernel(
-                script_uri, "main", kernel_platform, nullptr /* flags */,
+                script_uri, "main", kernel_platform, flags,
```
Comment (aam) [inline runtime/dart_init.cc]: Just curious - what changed so we want flags now?

File: frontend_server/lib/server.dart [hunk 2]
```diff
@@ -261,6 +246,14 @@ Future<int> starter(
   }
 
   if (options['train']) {
+    final String sdkRoot = options['sdk-root'];
```
Comment (mraleph) [inline frontend_server/lib/server.dart]: This chunk of code was reverted by @aam because it breaks the roll. 

You would need to keep this `if` empty.

Comment (mraleph) [reply frontend_server/lib/server.dart]: Disregard this comment. @jensjoha has fixed and relanded his snapshot training patch.

File: frontend_server/lib/server.dart [hunk 3]
```diff
@@ -298,9 +291,11 @@ Future<int> starter(
           boundaryKey = string.substring(RECOMPILE_INSTRUCTION_SPACE.length);
           state = _State.RECOMPILE_LIST;
         } else if (string == 'accept') {
-          compiler.acceptLastDelta();
+          // TODO(vegorov) remove this command from the protocol
         } else if (string == 'reject') {
-          compiler.rejectLastDelta();
+          // TODO(vegorov) evaluate conditions under which deltas can be
+          // rejected by the VM.
+          compiler.resetIncrementalCompiler();
```
Comment (a-siva) [inline frontend_server/lib/server.dart]: With the above suggested changes to acceptLastDelta and rejectLastDelta these changes would not be needed.

File: frontend_server/test/server_test.dart [hunk 4]
```diff
@@ -273,38 +274,6 @@ Future<int> main() async {
...
-      });
-      final int exitcode = await starter(args, compiler: compiler,
-        input: inputStreamController.stream,
-      );
-      expect(exitcode, equals(0));
-      inputStreamController.add('reject\n'.codeUnits);
-      await rejectCalled.first;
-      inputStreamController.close();
-    });
-
```
Comment (a-siva) [inline frontend_server/test/server_test.dart]: I guess if the methods are made nops then this test code doesn't have to be deleted either.

**Comment by Hixie** [comment]
The engine build is red, but the redness is a broken analyzer which will be fixed by this dart roll, so please disregard the red tree and land when ready.

File: frontend_server/lib/server.dart [hunk 5]
```diff
@@ -96,14 +96,6 @@ abstract class CompilerInterface {
...
   Future<Null> recompileDelta();
 
-  /// Accept results of previous compilation so that next recompilation cycle
-  /// won't recompile sources that were previously reported as changed.
-  void acceptLastDelta();
-
-  /// Reject results of previous compilation. Next recompilation cycle will
-  /// recompile sources indicated as changed.
-  void rejectLastDelta();
-
```
[the remark this answers is not in the document]
Comment (aam) [reply frontend_server/lib/server.dart]: We will look for the right solution as we fix dartbug.com/31919. I believe we could restore this functionality if it turns out to be at the right API for new incremental compiler

**Comment by aam** [comment]
Considering that this is keeping engine red, I will try to land it as soon as travis build is happy.

**Comment by aam** [comment]
Seeing some weird errors on travis as part of `flutter/travis/licenses.sh` run(which works locally for me):
```
Failed to precompile archive:tar:
'package:convert/src/hex/encoder.dart': malformed type: line 20 pos 13: cannot resolve class 'ChunkedConverter' from 'HexEncoder'
    extends ChunkedConverter<List<int>, String, List<int>, String> {
            ^
'package:convert/src/hex/encoder.dart': malformed type: line 20 pos 13: cannot resolve class 'ChunkedConverter' from 'HexEncoder'
    extends ChunkedConverter<List<int>, String, List<int>, String> {
```

**Comment by aam** [comment]
@Hixie @jason-simmons any ideas why `pub get` executed in `flutter/tools/licenses` pulls really old converter 1.1.1:

```
+flutter/travis/licenses.sh
Verifying license script is still happy...
Resolving dependencies... 
Overriding the upper bound Dart SDK constraint to <=2.0.0-dev.18.0 for the following packages:
archive, args, charcode, collection, convert, crypto, path, typed_data
To disable this you can set the PUB_ALLOW_PRERELEASE_SDK system environment variable to `false`, or you can silence this message by setting it to `quiet`.
< convert 1.1.1 (was 2.0.1) (2.0.1 available)
Downloading archive 1.0.33...
```

**Comment by aam** [comment]
> Seeing some weird errors on travis as part of flutter/travis/licenses.sh run(which works locally for me):

Seems to be caused by dartbug.com/31940, which is present in latest dart dev.18 that travis has picked up. Attempting to pin travis build to dev.16 to overcome the issue.

**Comment by Hixie** [comment]
(here's a link to that bug so that github does the cross-references: https://github.com/dart-lang/sdk/issues/31940)

File: frontend_server/lib/server.dart (hunk 5 above)
[the remark this answers is not in the document]
Comment (aam) [reply frontend_server/lib/server.dart]: Reintroduced methods back.

**Comment by aam** [comment]
@Hixie wrote
> The engine build is red, but the redness is a broken analyzer which will be fixed by this dart roll, so please disregard the red tree and land when ready.

With dart rolled the analyzer still fails seemingly the same way as before.

**Comment by aam** [comment]
dartanalyzer on the buildbot is still 1.25.0-dev.11.0 pulled from /b/build/slave/Linux_Engine/build/src/third_party/dart/tools/sdks/linux/dart-sdk/bin

**Comment by aam** [comment]
I don't see `gclient sync` executed as part of the build https://build.chromium.org/p/client.flutter/builders/Linux%20Engine/builds/2184

**Comment by mraleph** [comment]
I think it is part of bot_update: https://build.chromium.org/p/client.flutter/builders/Linux%20Engine/builds/2184/steps/bot_update/logs/stdio

**Comment by aam** [comment]
Aha, the download into `third_party/dart/tools/sdks` is done as part of `third_party/dart/DEPS` checked_in_dart_sdks. I wonder whether that is getting executed

**Comment by aam** [comment]
I think `third_party/dart/tools/sdks` is actually updated by https://github.com/flutter/buildroot/blob/master/tools/dart/update.py, which still has 1.25.0-dev.11.0/

**Comment by aam** [comment]
Opened https://github.com/flutter/buildroot/pull/123 to update `update.py`.

**Pull request merged**